### PR TITLE
Ensure ELMo can run on GPU

### PR DESCRIPTION
**Pull request by matt-peters**

### CODE REVIEW

File: allennlp/nn/util.py [hunk 1]
```diff
@@ -806,7 +806,7 @@ def add_sentence_boundary_token_ids(tensor: torch.Tensor,
         The new mask for the tensor, taking into account the appended tokens
         marking the beginning and end of the sentence.
     """
-    sequence_lengths = mask.sum(dim=1).data.numpy()
+    sequence_lengths = mask.sum(dim=1).data.cpu().numpy()
```
[the remark this answers is not in the document]
Comment (matt-peters) [reply allennlp/nn/util.py]: Did you benchmark this?  I'd prefer not to optimize the code until we confirm it's a bottleneck.

[the remark this answers is not in the document]
Comment (matt-gardner) [reply allennlp/nn/util.py]: I've benchmarked similar operations in my semantic parsing code, and yes, copying from GPU to CPU can kill your running time, if you do it frequently.  The more you can avoid it, the faster your code will be.  That said, though, if you avoid doing it frequently, I'm not certain whether one copy from GPU to CPU is better or worse than allocating a new tensor, calling scatter, cumsum, and the other operations here.

**Comment by matt-peters** [comment]
Right, which is why I think it's too early to spend any time optimizing this code... remember this function is just one piece of a huge network, so any effort to optimize should look at the entire network and then focus on the pieces that are the slowest.

**Comment by DeNeutoy** [comment]
Go ahead and merge this if you don't think it's slow - it only effects ELMo, so it's fairly isolated. If you do decide to merge in it's current state, please add a TODO with your name to profile this later. This will actually be straightfoward once we move to pytorch 0.3 as it has a built in profiler.

**Pull request merged**